### PR TITLE
test: added tests for enforcement notice

### PR DESCRIPTION
**Pull request by shilpabasu22**
## Ticket Number

https://pins-ds.atlassian.net/browse/AS-4017

## Checklist

- [ ] Requires infrastructure changes
- [ ] If adding or remove environment variables (e.g. in `docker-compose.yaml`) then I have updated the appropriate Helm chart
- [ ] I have updated the documentation accordingly
- [x] My commit history in this PR is linear
- [x] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `master` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

**Review by nssdesikan — APPROVED** [review]

File: e2e-tests/cypress/integration/before-you-start-enforcement-notice/before-you-start-enforcement-notice-steps.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import { verifyPageTitle } from '../../support/common/verify-page-title';
+import {
+  getEnforcementNoticeErrorMessage,
+  getEnforcementNoticeNo,
+  getEnforcementNoticeYes,
+} from '../../support/page-objects/enforcement-notice-po';
+import { getContinueButton, getErrorMessageSummary } from '../../support/page-objects/common-po';
+import { verifyErrorMessage } from '../../support/common/verify-error-message';
+const pageHeading = 'Have you received an enforcement notice?';
+const pageTitle = 'Have you received an enforcement notice? - Before you start - Appeal a householder planning decision - GOV.UK';
```
Comment (nssdesikan) [inline e2e-tests/cypress/integration/before-you-start-enforcement-notice/before-you-start-enforcement-notice-steps.js:13]: This title is as per the ticket but is this correct? (This is a common page for Householder and other planning decision)

**Review by nssdesikan — CHANGES_REQUESTED** [review]

**Review by swdpcomputing — APPROVED** [review]

File: packages/e2e-tests/cypress/integration/full-planning/before-you-start/enforcement-notice.feature [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  Scenario: AC03 appellant has not made any selection and they get an error message
+    Given appellant is on the enforcement notice page
+    When  appellant clicks on the continue button
+   Then appellant sees an error message 'Select yes if you have received an enforcement notice'
+
+  Scenario:AC04 Back Link
+    Given appellant is on the enforcement notice page
+    When appellant selects 'Yes' from the options
+    And appellant selects the back button
+    Then appellant is navigated to the type of planning application page
```
Comment (nssdesikan) [inline packages/e2e-tests/cypress/integration/full-planning/before-you-start/enforcement-notice.feature:27]: Change the text to _Any of following_

File: packages/e2e-tests/cypress/integration/full-planning/before-you-start/enforcement-notice/enforcement-notice-steps.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+Then('appellant gets navigated to the was your planning application granted or refused page',()=>{
+  cy.url().should('contain','/before-you-start/granted-or-refused');
+});
+
+Then('appellant is navigated to the shutter page',()=>{
+  cy.url().should('contain','/before-you-start/use-a-different-service');
+});
+
+Then('appellant is navigated to the type of planning application page',()=>{
```
Comment (nssdesikan) [inline packages/e2e-tests/cypress/integration/full-planning/before-you-start/enforcement-notice/enforcement-notice-steps.js:45]: Change the text to _Any of following

**Review by nssdesikan — CHANGES_REQUESTED** [review]

File: packages/e2e-tests/cypress/integration/full-planning/before-you-start/enforcement-notice.feature (hunk 2 above)
Comment (shilpabasu22) [reply packages/e2e-tests/cypress/integration/full-planning/before-you-start/enforcement-notice.feature]: updated as per review comments

File: packages/e2e-tests/cypress/integration/full-planning/before-you-start/enforcement-notice/enforcement-notice-steps.js (hunk 3 above)
Comment (shilpabasu22) [reply packages/e2e-tests/cypress/integration/full-planning/before-you-start/enforcement-notice/enforcement-notice-steps.js]: updated as per review comments

**Review by nssdesikan — APPROVED** [review]

**Pull request merged**